### PR TITLE
Code cleanup in stats

### PR DESCRIPTION
**Pull request by oscargus**
#### Brief description of what is fixed or changed
Some minor code quality improvements.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
LGTM

File: sympy/stats/drv.py [hunk 1]
```diff
@@ -123,7 +122,8 @@ def characteristic_function(self, t, **kwargs):
 
     @cacheit
     def compute_moment_generating_function(self, **kwargs):
-        x, t = symbols('x, t', real=True, finite=True, cls=Dummy)
+        t = Dummy('t', real=True)
+        x = Dummy('x', integer=True)
```
Comment (smichr) [inline sympy/stats/drv.py]: this does not match the old def...is that intentional?

Comment (oscargus) [reply sympy/stats/drv.py]: Correct. Discrete distributions. `x` is the index for the summation later, so I am quite sure that it is correct. Also compare with e.g. `compute_quantile` below. Probably one should change in the characteristics functions above as well (didn't notice those until now).

**Comment by oscarbenjamin** [comment]
Looks good.

**Pull request merged**